sdk/storage/azure-storage-common/src/file_io.cpp

```diff
@@ -30,15 +30,15 @@ namespace Azure { namespace Storage { namespace _internal {
 #if defined(AZ_PLATFORM_WINDOWS)
   FileReader::FileReader(const std::string& filename)
   {
-    int sizeNeeded
-        = MultiByteToWideChar(CP_UTF8, MB_ERR_INVALID_CHARS, filename.data(), -1, nullptr, 0);
+    int sizeNeeded = MultiByteToWideChar(
+        CP_UTF8, MB_ERR_INVALID_CHARS, filename.data(), int(filename.length()), nullptr, 0);
```

Comment:
Should we check the implied precondition that `filename.length() <= INT_MAX`? (Also on line 83.)

Reply:
I don't think filename longer than `INT_MAX` makes any sense on any OS or filesystem. So I don't think it's necessary

Reply:
The function doesn't have to succeed if that happens; you can check and then either abort or throw.